### PR TITLE
[WIP] `gx.get_context()` returns correct DataContext-type

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/data_context/test_get_data_context.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#         datasources={},
+#         stores={
+#             "expectations_store": {"class_name": "ExpectationsStore"},
+#             "evaluation_parameter_store": {"class_name": "EvaluationParameterStore"},
+#             "validation_result_store": {"class_name": "ValidationsStore"},
+#         },
+#         validations_store_name="validation_result_store",
+#         data_docs_sites={},
+#         validation_operators={},
+#     ), EphemeralDataContext)])
```
Comment (Shinnnyshinshin) [inline tests/data_context/test_get_data_context.py]: This is the parameterization that was giving me trouble... have some trouble passing in the class itself to check `isinstance()`

File: great_expectations/util.py [hunk 2]
```diff
@@ -1590,10 +1591,69 @@ def convert_ndarray_decimal_to_float_dtype(data: np.ndarray) -> np.ndarray:
...
-def get_context():
-    from great_expectations.data_context.data_context import DataContext
+def get_context(
+    context_root_dir: Optional[str] = None,
+    runtime_environment: Optional[dict] = None,
+    project_config: Optional[Any] = None,
+    ge_cloud_base_url: Optional[str] = None,
+    ge_cloud_access_token: Optional[str] = None,
+    ge_cloud_organization_id: Optional[str] = None,
+):
```
Comment (Kilo59) [inline great_expectations/util.py:1642]: Can we add a `DataContext` return type?

File: tests/data_context/test_get_data_context.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
+"""
+Why does this exist?
+
+"""
+import os
```
Comment (Kilo59) [inline tests/data_context/test_get_data_context.py:1]: Can we use `pathlib` exclusively and drop the use of `os.path`?

File: great_expectations/util.py [hunk 4]
```diff
@@ -1590,10 +1591,69 @@ def convert_ndarray_decimal_to_float_dtype(data: np.ndarray) -> np.ndarray:
...
 
-def get_context():
-    from great_expectations.data_context.data_context import DataContext
+def get_context(
+    context_root_dir: Optional[str] = None,
+    runtime_environment: Optional[dict] = None,
+    project_config: Optional[Any] = None,
+    ge_cloud_base_url: Optional[str] = None,
+    ge_cloud_access_token: Optional[str] = None,
+    ge_cloud_organization_id: Optional[str] = None,
```
Comment (Kilo59) [inline great_expectations/util.py:1641]: This is a good case for defining some function overloads.
https://mypy.readthedocs.io/en/stable/more_types.html?highlight=overloads#function-overloading

I'd love to pair on this 😄

**Comment by Shinnnyshinshin** [comment]
Closing as discussions with Don has resulted in a clearer understanding of the behavior of `gx.get_context()`

**Pull request closed without merging**